File: cflib_2_12/csharp/CFLib/CFLibArgumentOverflowException.cs
/*
 *  MSS Code Factory CFLib 2.11
 *
 *	Copyright (c) 2019 Mark Stephen Sobkow
 *	
 *	Licensed under the Apache License, Version 2.0 (the "License");
 *	you may not use this file except in compliance with the License.
 *	You may obtain a copy of the License at
 *	
 *		http://www.apache.org/licenses/LICENSE-2.0
 *	
 *	Unless required by applicable law or agreed to in writing, software
 *	distributed under the License is distributed on an "AS IS" BASIS,
 *	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *	See the License for the specific language governing permissions and
 *	limitations under the License.
 */

using System;
using System.Globalization;

namespace CFLib
{
	public class CFLibArgumentOverflowException : CFLibArgumentException {

		protected Object objValue = null;
		protected Object objMax = null;

		public CFLibArgumentOverflowException(
			String msg )
		: base(msg )
		{
		}

		public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			String msg )
		: base(throwingType, methName, msg)
		{
		}

		public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			String msg,
			Exception th )
		: base(throwingType, methName, msg, th)
		{
		}

		public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			String msg )
		: base(throwingType, methName, argNo, argName, msg)
		{
		}

		public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			String msg,
			Exception th )
		:	base(throwingType, methName, argNo, argName, msg, th)
		{
		}

		public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			short argValue,
			short maxValue )
		: base(throwingType, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatInt16( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatInt16(maxValue ) )
		{
			objValue = (Int16)argValue;
			objMax = (Int16)maxValue;
		}

		public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			short argValue,
			short maxValue,
			Exception th )
		: base(throwingType, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatInt16( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatInt16(maxValue ),
				th )
		{
            objValue = (Int16)argValue;
            objMax = (Int16)maxValue;
        }

        public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			int argValue,
			int maxValue )
		: base(throwingType, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatInt32( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatInt32(maxValue ) )
		{
            objValue = (Int32)argValue;
            objMax = (Int32)maxValue;
        }

        public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			int argValue,
			int maxValue,
			Exception th )
		: base(throwingType, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatInt32( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatInt32(maxValue ),
				th )
		{
            objValue = (Int32)argValue;
            objMax = (Int32)maxValue;
        }

        public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			long argValue,
			long maxValue )
		: base(throwingType, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatInt64( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatInt64(maxValue ) )
		{
            objValue = (Int64)argValue;
            objMax = (Int64)maxValue;
        }

        public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			long argValue,
			long maxValue,
			Exception th )
		: base(throwingType, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatInt64( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatInt64(maxValue ),
				th )
		{
            objValue = (Int64)argValue;
            objMax = (Int64)maxValue;
        }

        public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			float argValue,
			float maxValue )
		: base(throwingType, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatSingle( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatSingle(maxValue ) )
		{
			objValue = (Single)argValue;
			objMax = (Single)maxValue;
		}

		public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			float argValue,
			float maxValue,
			Exception th )
		: base(throwingType, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatSingle( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatSingle(maxValue ),
				th )
		{
            objValue = (Single)argValue;
            objMax = (Single)maxValue;
        }

        public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			double argValue,
			double maxValue )
		: base(throwingType, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatDouble( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatDouble(maxValue ) )
		{
            objValue = (Double)argValue;
            objMax = (Double)maxValue;
        }

        public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			double argValue,
			double maxValue,
			Exception th )
		: base(throwingType, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatDouble( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatDouble(maxValue ),
				th )
		{
            objValue = (Double)argValue;
            objMax = (Double)maxValue;
        }

        public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			DateTime argValue,
			DateTime maxValue )
		: base(throwingType, methName, argNo, argName,
				"value " + argValue.ToString()
					+ " must be no more than "
					+ maxValue.ToString() )
		{
			objValue = argValue;
			objMax = maxValue;
		}

		public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			DateTime argValue,
			DateTime maxValue,
			Exception th )
		: base(throwingType, methName, argNo, argName,
				"value " + argValue.ToString()
					+ " must be no more than "
					+ maxValue.ToString(),
				th )
		{
			objValue = argValue;
			objMax = maxValue;
		}

		public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			String argValue,
			String maxValue )
		: base(throwingType, methName, argNo, argName,
				"value " + argValue.ToString()
					+ " must be no more than "
					+ maxValue.ToString() )
		{
			objValue = argValue;
			objMax = maxValue;
		}

		public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			String argValue,
			String maxValue,
			Exception th )
		: base(throwingType, methName, argNo, argName,
				"value " + argValue.ToString()
					+ " must be no more than "
					+ maxValue.ToString(),
				th )
		{
			objValue = argValue;
			objMax = maxValue;
		}

		public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			BigDecimal argValue,
			BigDecimal maxValue )
		: base(throwingType, methName, argNo, argName,
				"value " + argValue.ToString()
					+ " must be no more than "
					+ maxValue.ToString() )
		{
			objValue = argValue;
			objMax = maxValue;
		}

		public CFLibArgumentOverflowException(
			Type throwingType,
			String methName,
			int argNo,
			String argName,
			BigDecimal argValue,
			BigDecimal maxValue,
			Exception th )
		: base(throwingType, methName, argNo, argName,
				"value " + argValue.ToString()
					+ " must be no more than "
					+ maxValue.ToString(),
				th )
		{
			objValue = argValue;
			objMax = maxValue;
		}

		public CFLibArgumentOverflowException(
			String fieldName,
			String msg )
		: base(fieldName, msg)
		{
		}

		public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			String msg )
		: base(fieldName, methName, msg)
		{
		}

		public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			String msg,
			Exception th )
		: base(fieldName, methName, msg, th)
		{
		}

		public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			String msg )
		: base(fieldName, methName, argNo, argName, msg)
		{
		}

		public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			String msg,
			Exception th )
		: base(fieldName, methName, argNo, argName, msg, th)
		{
		}

		public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			short argValue,
			short maxValue )
		: base(fieldName, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatInt16( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatInt16(maxValue ) )
		{
			objValue = (Int16)argValue;
			objMax = (Int16)maxValue;
		}

		public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			short argValue,
			short maxValue,
			Exception th )
		: base(fieldName, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatInt16( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatInt16(maxValue ),
				th )
		{
            objValue = (Int16)argValue;
            objMax = (Int16)maxValue;
        }

        public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			int argValue,
			int maxValue )
		: base(fieldName, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatInt32( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatInt32(maxValue ) )
		{
            objValue = (Int32)argValue;
            objMax = (Int32)maxValue;
        }

        public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			int argValue,
			int maxValue,
			Exception th )
		: base(fieldName, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatInt32( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatInt32(maxValue ),
				th )
		{
            objValue = (Int32)argValue;
            objMax = (Int32)maxValue;
        }

        public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			long argValue,
			long maxValue )
		: base(fieldName, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatInt64( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatInt64(maxValue ) )
		{
            objValue = (Int64)argValue;
            objMax = (Int64)maxValue;
        }

        public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			long argValue,
			long maxValue,
			Exception th )
		: base(fieldName, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatInt64( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatInt64(maxValue ),
				th )
		{
            objValue = (Int64)argValue;
            objMax = (Int64)maxValue;
        }

        public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			float argValue,
			float maxValue )
		: base(fieldName, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatSingle( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatSingle(maxValue ) )
		{
			objValue = (Single)argValue;
			objMax = (Single)maxValue;
		}

		public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			float argValue,
			float maxValue,
			Exception th )
		: base(fieldName, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatSingle( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatSingle(maxValue ),
				th )
		{
            objValue = (Single)argValue;
            objMax = (Single)maxValue;
        }

        public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			double argValue,
			double maxValue )
		: base(fieldName, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatDouble( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatDouble(maxValue ) )
		{
            objValue = (Double)argValue;
            objMax = (Double)maxValue;
        }

        public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			double argValue,
			double maxValue,
			Exception th )
		: base(fieldName, methName, argNo, argName,
				"value " + CFLibXmlUtil.formatDouble( argValue )
					+ " must be no more than "
					+ CFLibXmlUtil.formatDouble(maxValue ),
				th )
		{
            objValue = (Double)argValue;
            objMax = (Double)maxValue;
        }

        public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			DateTime argValue,
			DateTime maxValue )
		: base(fieldName, methName, argNo, argName,
				"value " + argValue.ToString()
					+ " must be no more than "
					+ maxValue.ToString() )
		{
			objValue = argValue;
			objMax = maxValue;
		}

		public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			DateTime argValue,
			DateTime maxValue,
			Exception th )
		: base(fieldName, methName, argNo, argName,
				"value " + argValue.ToString()
					+ " must be no more than "
					+ maxValue.ToString(),
				th )
		{
			objValue = argValue;
			objMax = maxValue;
		}

		public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			String argValue,
			String maxValue )
		: base(fieldName, methName, argNo, argName,
				"value " + argValue.ToString()
					+ " must be no more than "
					+ maxValue.ToString() )
		{
			objValue = argValue;
			objMax = maxValue;
		}

		public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			String argValue,
			String maxValue,
			Exception th )
		: base(fieldName, methName, argNo, argName,
				"value " + argValue.ToString()
					+ " must be no more than "
					+ maxValue.ToString(),
				th )
		{
			objValue = argValue;
			objMax = maxValue;
		}

		public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			BigDecimal argValue,
			BigDecimal maxValue )
		: base(fieldName, methName, argNo, argName,
				"value " + argValue.ToString()
					+ " must be no more than "
					+ maxValue.ToString() )
		{
			objValue = argValue;
			objMax = maxValue;
		}

		public CFLibArgumentOverflowException(
			String fieldName,
			String methName,
			int argNo,
			String argName,
			BigDecimal argValue,
			BigDecimal maxValue,
			Exception th )
		: base(fieldName, methName, argNo, argName,
				"value " + argValue.ToString()
					+ " must be no more than "
					+ maxValue.ToString(),
				th )
		{
			objValue = argValue;
			objMax = maxValue;
		}
		
		public Object getValue() {
			return( objValue );
		}
		
		public Object getMax() {
			return( objMax );
		}
	}
}

File: cflib_2_12/cplus/libcflib/CFLibArgumentOverflowException.cpp
/*
 *  MSS Code Factory CFLib 2.12
 *
 *	Copyright (c) 2020 Mark Stephen Sobkow
 *	
 *	Licensed under the Apache License, Version 2.0 (the "License");
 *	you may not use this file except in compliance with the License.
 *	You may obtain a copy of the License at
 *	
 *	    http://www.apache.org/licenses/LICENSE-2.0
 *	
 *	Unless required by applicable law or agreed to in writing, software
 *	distributed under the License is distributed on an "AS IS" BASIS,
 *	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *	See the License for the specific language governing permissions and
 *	limitations under the License.
 */

#include <cflib/CFLib.hpp>
#include <cflib/CFLibXmlUtil.hpp>
#include <cflib/CFLibArgumentException.hpp>
#include <cflib/CFLibArgumentOverflowException.hpp>

namespace cflib {

	const std::string CFLibArgumentOverflowException::S_MUST_BE_NO_MORE_THAN( " must be no more than " );

	CFLibArgumentOverflowException::~CFLibArgumentOverflowException() {
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const CFLibArgumentOverflowException& src )
	: std::overflow_error( src )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& msg )
	: std::overflow_error( msg )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		const std::string& msg )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE + msg )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		int argNo,
		const std::string& argName,
		const std::string& msg )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE_ARGUMENT + std::to_string( argNo ) + CFLib::S_SPACE + argName + CFLib::S_SPACE + msg )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		int argNo,
		const std::string& argName,
		int16_t argValue,
		int16_t minValue )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE_ARGUMENT + std::to_string( argNo ) + CFLib::S_SPACE + argName + CFLib::S_SPACE + CFLib::S_VALUE_SPACE + std::to_string( argValue ) + S_MUST_BE_NO_MORE_THAN + std::to_string( minValue ) )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		int argNo,
		const std::string& argName,
		int32_t argValue,
		int32_t minValue )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE_ARGUMENT + std::to_string( argNo ) + CFLib::S_SPACE + argName + CFLib::S_SPACE + CFLib::S_VALUE_SPACE + std::to_string( argValue ) + S_MUST_BE_NO_MORE_THAN + std::to_string( minValue ) )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		int argNo,
		const std::string& argName,
		int64_t argValue,
		int64_t minValue )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE_ARGUMENT + std::to_string( argNo ) + CFLib::S_SPACE + argName + CFLib::S_SPACE + CFLib::S_VALUE_SPACE + std::to_string( argValue ) + S_MUST_BE_NO_MORE_THAN + std::to_string( minValue ) )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		int argNo,
		const std::string& argName,
		uint16_t argValue,
		uint16_t minValue )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE_ARGUMENT + std::to_string( argNo ) + CFLib::S_SPACE + argName + CFLib::S_SPACE + CFLib::S_VALUE_SPACE + std::to_string( argValue ) + S_MUST_BE_NO_MORE_THAN + std::to_string( minValue ) )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		int argNo,
		const std::string& argName,
		uint32_t argValue,
		uint32_t minValue )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE_ARGUMENT + std::to_string( argNo ) + CFLib::S_SPACE + argName + CFLib::S_SPACE + CFLib::S_VALUE_SPACE + std::to_string( argValue ) + S_MUST_BE_NO_MORE_THAN + std::to_string( minValue ) )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		int argNo,
		const std::string& argName,
		uint64_t argValue,
		uint64_t minValue )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE_ARGUMENT + std::to_string( argNo ) + CFLib::S_SPACE + argName + CFLib::S_SPACE + CFLib::S_VALUE_SPACE + std::to_string( argValue ) + S_MUST_BE_NO_MORE_THAN + std::to_string( minValue ) )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		int argNo,
		const std::string& argName,
		float argValue,
		float minValue )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE_ARGUMENT + std::to_string( argNo ) + CFLib::S_SPACE + argName + CFLib::S_SPACE + CFLib::S_VALUE_SPACE + std::to_string( argValue ) + S_MUST_BE_NO_MORE_THAN + std::to_string( minValue ) )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		int argNo,
		const std::string& argName,
		double argValue,
		double minValue )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE_ARGUMENT + std::to_string( argNo ) + CFLib::S_SPACE + argName + CFLib::S_SPACE + CFLib::S_VALUE_SPACE + std::to_string( argValue ) + S_MUST_BE_NO_MORE_THAN + std::to_string( minValue ) )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		int argNo,
		const std::string& argName,
		const std::chrono::system_clock::time_point& argValue,
		const std::chrono::system_clock::time_point& minValue )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE_ARGUMENT + std::to_string( argNo ) + CFLib::S_SPACE + argName + CFLib::S_SPACE + CFLib::S_VALUE_SPACE + CFLibXmlUtil::formatTimestamp( argValue ) + S_MUST_BE_NO_MORE_THAN + CFLibXmlUtil::formatTimestamp( minValue ) )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		int argNo,
		const std::string& argName,
		const std::string& argValue,
		const std::string& minValue )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE_ARGUMENT + std::to_string( argNo ) + CFLib::S_SPACE + argName + CFLib::S_SPACE + CFLib::S_VALUE_SPACE + argValue + S_MUST_BE_NO_MORE_THAN + minValue )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		int argNo,
		const std::string& argName,
		mpfr_srcptr argValue,
		mpfr_srcptr minValue )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE_ARGUMENT + std::to_string( argNo ) + CFLib::S_SPACE + argName + CFLib::S_SPACE + CFLib::S_VALUE_SPACE + CFLibXmlUtil::formatMPFR( argValue ) + S_MUST_BE_NO_MORE_THAN + CFLibXmlUtil::formatMPFR( minValue ) )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& throwingClassName,
		const std::string& methName,
		int argNo,
		const std::string& argName,
		const CFLibGenericBigDecimal& argValue,
		const CFLibGenericBigDecimal& minValue )
	: std::overflow_error( throwingClassName + CFLib::S_DOUBLE_COLON + methName + CFLib::S_PAREN_SPACE_ARGUMENT + std::to_string( argNo ) + CFLib::S_SPACE + argName + CFLib::S_SPACE + CFLib::S_VALUE_SPACE + argValue.toString() + S_MUST_BE_NO_MORE_THAN + minValue.toString() )
	{
	}

	CFLibArgumentOverflowException::CFLibArgumentOverflowException(
		const std::string& fieldName,
		const std::string& msg )
	: std::overflow_error( fieldName + CFLib::S_SPACE + msg )
	{
	}

}
